p2p/message.go

```diff
@@ -273,6 +279,11 @@ func (bm *BlockRequestMessage) Decode(r io.Reader) error {
 	return nil
 }
 
+// Id Returns the ID of the block
+func (bm *BlockRequestMessage) Id() string {
+	return string(bm.ID)
```

Comment:
I think we should also probably encode+hash the entire message for BlockRequest and BlockResponse as well, since there's a chance of people using duplicate IDs

Reply:
We're assuming the ID is a random number that is unique

Reply:
okay, I'm still unsure about this since I think the chance of duplicate IDs is pretty high. I guess this is ok for now but we should probably change it to the hash of the entire message as well, or clear the cache of seen messages after some time